doc/en/components/grids/_shared/row-drag.md

```diff
@@ -0,0 +1,798 @@
+        </div>
+    </ng-template>
+    <igx-row-island>
+        <ng-template IgxRowDragGhost>
+            <img src="smile.gif" height="42" width="42">
+        </ng-template>
+    </igx-row-island>
+</{ComponentSelector}>
+```
+<!-- ComponentEnd: HierarchicalGrid -->
+
+<code-view style="height:600px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-multi-row-drag"
+           github-src="{ComponentSample}/multi-row-drag"
+           alt="{Platform} {ComponentName} Multi Row Drag">
+</code-view>
+
+### Templating the drag icon
+The drag handle icon can be templated using the grid's `DragIndicatorIconTemplate`. In the example we're building, let's change the icon from the default one (**drag_indicator**) to **drag_handle**.
+To do so, we can use the `IgxDragIndicatorIcon` to pass a template inside of the `{ComponentSelector}`'s body:
+```html
+<{ComponentSelector}>
+    <ng-template igxDragIndicatorIcon>
+        <igx-icon>drag_handle</igx-icon>
+    </ng-template>
+</{ComponentSelector}>
+```
+
+Once we've set the new icon template, we also need to adjust the **DEFAULT** icon in our **DragIcon enum**, so it's properly change by the `ChangeIcon` method:
+```typescript
+enum DragIcon {
+    DEFAULT = "drag_handle",
+}
+```
+<!-- ComponentStart: TreeGrid, HierarchicalGrid -->
+### Styling the drop area
+Once our drop handlers are properly configured, all that's left is to style our drop area a bit:
+```css
+.drop-area {
+    width: 160px;
+    height: 160px;
+    background-color: #d3d3d3;
+    border: 1px dashed #131313;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-flow: column;
+    text-align: center;
+    margin: 8px;
+}
+
+:host {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-flow: column;
+    width: 100%;
+}
+```
+
+The result can be seen in the demo below:
+<!-- ComponentEnd: TreeGrid, HierarchicalGrid -->
+<!-- ComponentStart: Grid -->
+Once our drop handlers are properly configured, we're good to go!
+The result of the configuration can be seem below:
+<!-- ComponentEnd: Grid -->
+
+#### Example Demo
+<!-- ComponentStart: Grid -->
+
+<code-view style="height:550px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-row-drag-to-grid"
+           github-src="{ComponentSample}/row-drag-to-grid"
+           alt="{Platform} {ComponentName} Row Drag">
+</code-view>
+
+
+<!-- ComponentEnd: Grid -->
+
+<!-- ComponentStart: TreeGrid, HierarchicalGrid -->
+
+<code-view style="height:560px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-row-drag"
+           github-src="{ComponentSample}/row-drag"
+           alt="{Platform} {ComponentName} Row Drag">
+</code-view>
+<!-- ComponentEnd: TreeGrid, HierarchicalGrid -->
```

Comment:
replace above 3 code-view blocks with just one code-view and remove component build flag because routing path to the sample will be the same for all grid samples in Sample Browser.